src/core/referenceResolver.ts

```diff
@@ -95,7 +96,18 @@ export default class ReferenceResolver {
     }
 
     public async registerRemoteSchema(url: string): Promise<void> {
-        const res = await fetch(url);
+        const fetchOptions:any = {};
+        if (process.env.HTTP_PROXY) {
+            const proxyUrl = new URL(process.env.HTTP_PROXY);
+            const agentOptions:any = {};
+            agentOptions.host = proxyUrl.hostname;
+            agentOptions.port = proxyUrl.port;
```

Comment:
Isn't it necessary to set the `agentOptions.protocol` property?

Reply:
Hi @horiuchi, you are correct.  The following two commits add support for the `HTTPS_PROXY` / `NO_PROXY` environment variables + fix the `agentOptions.protocol` setting.

Reply:
Thank you for your fixing of the proxy configuration.